src/js/media-controller.js

```diff
@@ -219,39 +225,71 @@ class MediaController extends MediaContainer {
       const {
         mediaEvents,
         rootEvents,
-        trackListEvents
+        textTracksEvents,
       } = MediaUIStates[key];
 
       const handler = this._mediaStatePropagators[key];
 
-      mediaEvents?.forEach((eventName)=>{
+      mediaEvents?.forEach((eventName) => {
         media.removeEventListener(eventName, handler);
       });
 
-      rootEvents?.forEach((eventName)=>{
+      rootEvents?.forEach((eventName) => {
         this.getRootNode().removeEventListener(eventName, handler);
       });
 
-      trackListEvents?.forEach((eventName)=>{
+      textTracksEvents?.forEach((eventName) => {
         media.textTracks?.removeEventListener(eventName, handler);
       });
     });
 
+    media.videoTracks?.removeEventListener('addtrack', this.#handleVideoTrack);
+    media.videoTracks?.removeEventListener('change', this.#handleVideoTrack);
+    media.videoTracks?.removeEventListener('removetrack', this.#handleVideoTrack);
+
+    this.#handleVideoTrack({ type: 'removeall' });
+
     // Reset to paused state
     // TODO: Can we just reset all state here?
     // Should hasPlayed refer to the media element or the controller?
     // i.e. the poster might re-show if not handled by the poster el
     this.propagateMediaState(MediaUIProps.MEDIA_PAUSED, true);
   }
 
+  #handleVideoTrack = ({ type = '', track = null } = {}) => {
```

Comment:
This is complicated. It'd be nice if we could pull this into the above logic more, though I see how it's different because there needs to be a listener on each individual track.

Reply:
yes, that's what I was hinting at in point 1 in the PR description.
I don't like it that much either.